packages/gcc10.rb

```diff
@@ -69,128 +76,338 @@ def self.build
     # Fix "crtbeginT.o: relocation R_X86_64_32 against hidden symbol `__TMC_END__' can not be used when making a shared object"
     # when building static llvm
     system "sed -i 's/-fbuilding-libgcc -fno-stack-protector/-fbuilding-libgcc -fPIC -fno-stack-protector/g' libgcc/Makefile.in"
-    Dir.chdir("objdir") do
-      ENV['CFLAGS'] = '-fPIC'
-      ENV['CXXFLAGS'] = '-fPIC'
+    Dir.chdir('objdir') do
+      @cflags = '-fPIC'
+      @cxxflags = '-fPIC'
+      @languages = 'c,c++,jit,objc,fortran,go'
       case ARCH
       when 'armv7l', 'aarch64'
-        system "../configure #{CREW_OPTIONS} \
-        --enable-checking=release \
-        --disable-multilib \
-        --enable-lto \
-        --enable-threads=posix \
-        --disable-werror \
-        --disable-libmpx \
-        --enable-static \
-        --enable-shared \
-        --with-pic \
-        --program-suffix=-#{gcc_version} \
-        --with-arch=armv7-a \
-        --with-tune=cortex-a15 \
-        --with-fpu=neon \
-        --with-float=hard \
-        --with-build-config=bootstrap-lto-lean"
+        system "env CFLAGS=#{@cflags} CXXFLAGS=#{@cxxflags} \
+           LD_LIBRARY_PATH=#{CREW_LIB_PREFIX} \
+           LIBRARY_PATH=#{CREW_LIB_PREFIX} PATH=#{@path} \
+          ../configure #{CREW_OPTIONS} \
+          --disable-libmpx \
+          --disable-libssp \
+          --disable-multilib \
+          --disable-werror \
+          --enable-cet=auto \
+          --enable-checking=release \
+          --enable-clocale=gnu \
+          --enable-default-pie \
+          --enable-default-ssp \
+          --enable-gnu-indirect-function \
+          --enable-gnu-unique-object \
+          --enable-host-shared \
+          --enable-languages=#{@languages} \
+          --enable-lto \
+          --enable-plugin \
+          --enable-shared \
+          --enable-static \
+          --enable-threads=posix \
+          --program-suffix=-#{gcc_version} \
+          --with-arch=armv7-a \
+          --with-build-config=bootstrap-lto-lean \
+          --with-float=hard \
+          --with-fpu=neon \
+          --with-isl \
+          --with-pic \
+          --with-system-zlib \
+          --with-tune=cortex-a15"
       when 'x86_64'
-        system "../configure  #{CREW_OPTIONS} \
-        --enable-checking=release \
-        --enable-lto \
-        --disable-multilib \
-        --enable-threads=posix \
-        --disable-werror \
-        --disable-libmpx \
-        --enable-static \
-        --enable-shared \
-        --with-pic \
-        --program-suffix=-#{gcc_version} \
-        --with-arch-64=x86-64 \
-        --with-build-config=bootstrap-lto-lean"
+        system "env CFLAGS=#{@cflags} CXXFLAGS=#{@cxxflags}  \
+          LD_LIBRARY_PATH=#{CREW_LIB_PREFIX} \
```

Comment:
This line can be removed since it is exported already

Reply:
without LD_LIBRARY_PATH being exported, ruby won't even run so you don't have to worry about it.

Reply:
Thanks all, That code is a kludge from when I wasn't sure if the environment variables were being passed correctly.